src/online/scoreboard-system/createScoreboardClient.js

```diff
@@ -209,46 +208,39 @@ export default function createScoreboardClient({
     entry ${ENTRY}
   }`
 
-  const scoreboardClient = {
-    signUp({ username, password, email }) {
+  return {
+    async signUp({ username, password, email }) {
       invariant(typeof username === 'string', 'username must be a string')
       invariant(typeof password === 'string', 'password must be a string')
       invariant(typeof email === 'string', 'email must be a string')
-      // TODO [#632]: Convert the `signUp` method to async function (instead of using `co`) in src/online/scoreboard-system/createScoreboardClient.js
-      // See issue #575 for more details.
-      return co(function*() {
-        const { idToken } = yield* authenticationFlow.signUp(
```

Comment:
This one is a bit tricky because we’re using `yield*` instead of `yield`. Simply changing it to `await` doesn't work as we will get back an iterator instead of a Promise. To really fix this you have to convert generator functions in [`authenticationFlow`](https://github.com/bemusic/bemuse/blob/master/src/online/scoreboard-system/authenticationFlow.js) to async functions. Would you help do that for me?